src/planning/recipe-index.ts

```diff
@@ -98,9 +98,11 @@ export class RecipeIndex {
 
   constructor(arc: Arc, {reportGenerations = false} = {}) {
     const trace = Tracing.start({cat: 'indexing', name: 'RecipeIndex::constructor', overview: true});
+    // TODO: Inject an IdGenerator instead of starting a new session here.
```

Comment:
Why? What's wrong with leaving the creation of a session to the Arc constructor? I guess you can argue that it should be testable, though I would be fine with leaving it as is and removing a TODO.

Reply:
This is the RecipeIndex constructor, not the Arc constructor. I don't really know if creating a new RecipeIndex is analogous to starting a new session, I'll defer to you here.

Reply:
I see, I missed tis. Yeah... either way is fine. :)